### ISSUE TITLE
🐛 git add -p added lines are shown empty

### ISSUE THREAD

**Issue by domrim** [dandavison/delta#557]
since the last update `git add -p` does not show the content of new lines. see screenshot.

![image](https://user-images.githubusercontent.com/15873147/113883337-59bbea80-97be-11eb-8fbd-eae2a7d6d147.png)

my relevant `.gitconfig` parts, all pulled from the Readme:
```
[core]
    pager = delta

[interactive]
    diffFilter = delta --color-only

[delta]
    features = side-by-side line-numbers decorations
    whitespace-error-style = 22 reverse

[delta "decorations"]
    commit-decoration-style = bold yellow box ul
    file-style = bold yellow ul
     file-decoration-style = none
```

running delta v0.7.1 on macOS 11.2.3

**Comment by domrim** [comment]
funny: i just checked it with another repo. there it is fine.

could it depend on the file content?

**Comment by domrim** [comment]
yes, i've created a small minimal project which triggers the bug for me. you can find it here: https://github.com/domrim/example

**Comment by dandavison** [comment]
Thanks! I'll take a look.

**Comment by CaioCSdev** [comment]
I'm having the exact same issue. git diff works, but git add -p fails :thinking: 

![image](https://user-images.githubusercontent.com/1172532/118757121-c9ec7000-b842-11eb-886b-f443b65f6271.png)

**Comment by CaioCSdev** [comment]
I discovered the issue!! :tada: 

Is the line break char.
Delta works perfectly with `LF`, but when the file uses `CRFL` this strange error happens, and only on `git add -p` (at least so far)

got a hint because I was using cshap

**Comment by dandavison** [comment]
Hi @CaioCSdev, awesome, thanks! Now that you point this out I realise that the test suite has no coverage for alternate line endings. This sounds very fixable now.

**Comment by michaelblyons** [comment]
Is this done with #664 or waiting for #754?

**Comment by dandavison** [comment]
Thanks @michaelblyons. This should be fixed already by #664 (please someone shout out if that's not true). #754 raises a related question about that fix that we don't yet have an answer to, but the #664 fix still stands currently.
